bring/src/main/java/com/bobocode/petros/bring/factory/postprocessor/FieldInjectionBeanPostProcessor.java

```diff
@@ -0,0 +1,39 @@
+package com.bobocode.petros.bring.factory.postprocessor;
+
+import com.bobocode.petros.bring.annotation.Autowired;
+import com.bobocode.petros.bring.context.ApplicationContext;
+import com.bobocode.petros.bring.context.domain.BeanReference;
+import com.bobocode.petros.bring.exception.NoUniqueBeanException;
+import lombok.AccessLevel;
+import lombok.Setter;
+import lombok.SneakyThrows;
+
+import java.lang.reflect.Field;
+
+public class FieldInjectionBeanPostProcessor implements BeanPostProcessor {
+
+    @Setter(AccessLevel.PACKAGE)
+    private ApplicationContext context;
+
+    //TODO: make possible to inject collection of beans, new method in ApplicationContext that return all beans by bean type ?
+    @Override
+    @SneakyThrows
+    public BeanReference postProcess(BeanReference beanReference) {
+        var beanObject = beanReference.getBeanObject();
+        for (Field field : beanObject.getClass()
+                .getDeclaredFields()) {
+            if (field.isAnnotationPresent(Autowired.class)) {
+                field.setAccessible(true);
+                Object objectToAutowire;
+                try {
+                    objectToAutowire = context.getBean(field.getType());
+                } catch (NoUniqueBeanException exception) {
```

Comment:
Could you please point me to the case when this exception is supposed to be thrown during an invocation of context.getBean(field.getType()) method? I've checked current implementation of it and this exception is not expected to be thrown.